core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -515,6 +515,11 @@ class ReplicaManager(val config: KafkaConfig,
             case Some(p) =>
               if (p eq ReplicaManager.OfflinePartition)
                 throw new KafkaStorageException("Partition %s is in an offline log directory on broker %d".format(topicPartition, localBrokerId))
+              p.leaderReplicaIfLocal match {
+                case Some(_) =>
```

Comment:
You should be able to use `case Some(_) => p` and thus omit the `p` that's currently the last expression in the block.